src/service/dashboards/timed_annotations.rs

```diff
@@ -0,0 +1,234 @@
+
+    #[cfg(feature = "enterprise")]
+    for timed_annotation in timed_annotations_res.iter() {
+        match super_cluster::emit_timed_annotation_create_event(
+            dashboard_id,
+            timed_annotation.clone(),
+            true,
+        )
+        .await
+        {
+            Ok(_) => (),
+            Err(e) => tracing::error!(
+                "[dashboard_id: {}] Failed to emit event to super cluster: {:?}",
+                dashboard_id,
+                e
+            ),
+        };
+    }
+
+    Ok(timed_annotations_res)
+}
+
+#[tracing::instrument]
+pub async fn get_timed_annotations(
+    dashboard_id: &str,
+    panels: Option<Vec<String>>,
+    start_time: i64,
+    end_time: i64,
+) -> Result<Vec<TimedAnnotation>, anyhow::Error> {
+    let annotations =
+        table::timed_annotations::get(dashboard_id, panels, start_time, end_time).await?;
+    Ok(annotations)
+}
+
+#[tracing::instrument]
+pub async fn delete_timed_annotations(
+    dashboard_id: &str,
+    req: TimedAnnotationDelete,
+) -> Result<(), anyhow::Error> {
+    if req.annotation_ids.is_empty() {
+        return Err(anyhow::anyhow!("annotation_ids cannot be empty"));
+    }
+    table::timed_annotations::delete_many(dashboard_id, &req.annotation_ids).await?;
+
+    #[cfg(feature = "enterprise")]
+    for id in req.annotation_ids.iter() {
+        match super_cluster::emit_timed_annotation_delete_event(dashboard_id, id).await {
+            Ok(_) => (),
+            Err(e) => tracing::error!(
+                "[timed_annotation_id: {}] Failed to emit event to super cluster: {:?}",
+                id,
+                e
+            ),
+        };
+    }
+
+    Ok(())
+}
+
+#[tracing::instrument]
+pub async fn update_timed_annotations(
+    dashboard_id: &str,
+    timed_annotation_id: &str,
+    req: &TimedAnnotationUpdate,
+) -> Result<TimedAnnotation, anyhow::Error> {
+    table::timed_annotations::update(dashboard_id, timed_annotation_id, req.clone()).await?;
+    #[cfg(feature = "enterprise")]
+    super_cluster::emit_timed_annotation_put_event(dashboard_id, timed_annotation_id, req.clone())
+        .await?;
+
+    if let Some(new_panels) = &req.panels {
+        let existing_panels =
+            table::timed_annotation_panels::get_panels(timed_annotation_id).await?;
+        let panels_to_add: Vec<String> = new_panels
+            .iter()
+            .filter(|panel| !existing_panels.contains(panel))
+            .cloned()
+            .collect();
+
+        // Add only new panels
+        if !panels_to_add.is_empty() {
+            table::timed_annotation_panels::insert_many_panels(
+                timed_annotation_id,
+                panels_to_add.clone(),
+            )
+            .await?;
+            #[cfg(feature = "enterprise")]
+            super_cluster::emit_timed_annotation_panels_put_event(
+                timed_annotation_id,
+                panels_to_add,
+            )
+            .await?;
+        }
```

Comment:
This just adds the annotations to panels when `req.panels` contains panel IDs that the annotation is not currently associated with.

Should this also remove the annotation from panels when `req.panels` does **not** contain panel IDs that the annotation is already associated with? Or is this function indended to by "add only".

Reply:
If this function is intended to be "add only" then I recommend renaming it to something that reflects that behavior such as `add_timed_annotation_to_panels`.

When I read the current name `update_timed_annotations` the name makes me expect that this function will synchronize the associated panels to match the panels specified in `req.panels` and that it will perform both additions and deletions to do the synchronization.